feature/channel/src/main/java/com/m3u/feature/channel/ChannelViewModel.kt

```diff
@@ -81,6 +85,23 @@ class ChannelViewModel @Inject constructor(
     internal val channel: StateFlow<Channel?> = playerManager.channel
     internal val playlist: StateFlow<Playlist?> = playerManager.playlist
 
+    val adjacentChannels: StateFlow<AdjacentChannels?> = flatmapCombined(
+        playlist.filterNotNull(),
+        channel.filterNotNull()
```

Comment:
Don't use `filterNotNull` here. If these upstream flows emit null it means they are cleared. So the adjacentChannels should be cleared as well.

Reply:
The issue occurs in the usage of flatmapCombined, where the expected input type is Any, but a value of type Playlist? is being passed.
![image](https://github.com/user-attachments/assets/808c845c-5a8d-4fdb-a9b6-45e9a7bb6008)

Reply:
Maybe we should adjust the method again.